### PR TITLE
Add Security Warning Bar for untrusted notebooks

### PR DESCRIPTION
**Pull request by gitanuj**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/970)

1. Mark all notebooks downloaded from the Public Gallery as untrusted (by setting "untrusted" flag in notebook metadata)
2. Block executing Untrusted notebook from Keyboard shortcut, Hover button, Context Menu, and Command bar menu
3. Allow a user to Trust Notebook

Addresses this bug - https://msdata.visualstudio.com/CosmosDB/_workitems/edit/1224092/

image

### CODE REVIEW

File: src/Explorer/Tabs/NotebookV2Tab.ts [hunk 1]
```diff
@@ -84,11 +85,13 @@ export default class NotebookTabV2 extends NotebookTabBase {
 
   protected getTabsButtons(): CommandButtonComponentProps[] {
     const availableKernels = NotebookTabV2.clientManager.getAvailableKernelSpecs();
+    const isNotebookUntrusted = this.notebookComponentAdapter.isNotebookUntrusted();
+
+    const runBtnTooltip = isNotebookUntrusted ? NotebookUtil.UntrustedNotebookRunHint : undefined;
```
Comment (swviswan) [inline src/Explorer/Tabs/NotebookV2Tab.ts:90]: >UntrustedNotebookRunHint [](http://example.com/codeflow?start=61&length=24)

dont see this work in the preview url

File: src/Utils/GalleryUtils.ts [hunk 2]
```diff
@@ -243,6 +243,11 @@ export function downloadItem(
         const notebook = JSON.parse(response.data) as Notebook;
         removeNotebookViewerLink(notebook, data.newCellId);
 
+        if (!data.isSample) {
+          const metadata = notebook.metadata as { [name: string]: unknown };
+          metadata.untrusted = true;
```
Comment (swviswan) [inline src/Utils/GalleryUtils.ts:248]: >   metadata.untrusted = true; [](http://example.com/codeflow?start=7&length=29)

thoughts on this being on the server?

**Review by swviswan — APPROVED** [review]
:shipit:

File: src/Explorer/Tabs/NotebookV2Tab.ts (hunk 1 above)
Comment (gitanuj) [reply src/Explorer/Tabs/NotebookV2Tab.ts:90]: You can hover over the play buttons and see this tooltip.

https://user-images.githubusercontent.com/693092/128913241-b341e0f9-f0ea-41a8-9ab2-9b8a500d685b.mp4

File: src/Utils/GalleryUtils.ts (hunk 2 above)
Comment (gitanuj) [reply src/Utils/GalleryUtils.ts:248]: We could also do this at the time of publishing server side. I don't see any down sides to that either.

**Pull request merged**